core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -368,6 +371,8 @@ private DefaultFunctionResolver dayOfWeek() {
    */
   private DefaultFunctionResolver dayOfYear(BuiltinFunctionName dayOfYear) {
     return define(dayOfYear.getName(),
+        implWithProperties((functionProperties, arg) -> DateTimeFunction.dayOfYearToday(
+            functionProperties.getQueryStartClock()), INTEGER, TIME),
```

Comment:
```suggestion
        implWithProperties(nullMissingHandlingWithProperties((functionProperties, arg) -> DateTimeFunction.dayOfYearToday(
            functionProperties.getQueryStartClock())), INTEGER, TIME),
```

Reply:
Fixed in 6465ec867cb5a13d21058604391687fe7cc03cb5